File: Client-Admin/src/Models/Order.ts
import { PageObject } from "./Common";
import { VariantGet } from "./Variant";

export type OrderResponse = {
    items: OrderGet[];
    page: PageObject;
}

export type OrderGet = {
    orderId: number;
    customerName: string;
    orderDate: string;
    totalAmount: number;
    status: string | null;
    address: string;    
    phone: string | null;
    paymentMethod: string | null;
    shippingService: string | null;
    notes: string | null;
    orderDetails: OrderDetailGet[];
}

export type OrderDetailGet = {
    orderDetailId: number;
    orderId: number;
    variant: VariantGet;
    quantity: number;
    price: number;
}

export interface OrderStatusPost {
    statusOrder: string
  }

File: Server/Models/Order.cs
﻿using System;
using System.Collections.Generic;

namespace Server.Models;

public partial class Order
{
    public int OrderId { get; set; }

    public int CustomerId { get; set; }

    public DateTime OrderDate { get; set; }

    public decimal TotalAmount { get; set; }

    public string? Status { get; set; }

    public string Address { get; set; } = null!;

    public string? Phone { get; set; }

    public string? PaymentMethod { get; set; }

    public string? Notes { get; set; }

    public virtual Customer Customer { get; set; } = null!;

    public virtual ICollection<OrderDetail> OrderDetails { get; set; } = new List<OrderDetail>();
}
